From a64bac5fc75aa2872a8abe54a9f9aaa0fce31588 Mon Sep 17 00:00:00 2001
From: Kurt Mahan <kmahan@freescale.com>
Date: Thu, 22 May 2008 09:50:31 -0600
Subject: [PATCH] Add elf relocation types for TLS/NPTL.

These relocations are from the draft TLS/NPTL ABI document v0.2
by Joseph Myers of CodeSourcery.

LTIBName: mcfv4e-tlsnptl-elf-reloc
Signed-off-by: Kurt Mahan <kmahan@freescale.com>
---
 include/asm-m68k/elf.h |   20 ++++++++++++++++++++
 1 files changed, 20 insertions(+), 0 deletions(-)

--- a/include/asm-m68k/elf.h
+++ b/include/asm-m68k/elf.h
@@ -34,6 +34,26 @@
 #define R_68K_GLOB_DAT	20
 #define R_68K_JMP_SLOT	21
 #define R_68K_RELATIVE	22
+/* TLS static relocations */
+#define	R_68K_TLS_GD32		25
+#define	R_68K_TLS_GD16		26
+#define	R_68K_TLS_GD8		27
+#define	R_68K_TLS_LDM32		28
+#define	R_68K_TLS_LDM16		29
+#define	R_68K_TLS_LDM8		30
+#define	R_68K_TLS_LDO32		31
+#define	R_68K_TLS_LDO16		32
+#define	R_68K_TLS_LDO8		33
+#define	R_68K_TLS_IE32		34
+#define	R_68K_TLS_IE16		35
+#define	R_68K_TLS_IE8		36
+#define	R_68K_TLS_LE32		37
+#define	R_68K_TLS_LE16		38
+#define	R_68K_TLS_LE8		39
+/* TLS dynamic relocations */
+#define	R_68K_TLS_DTPMOD32	40
+#define	R_68K_TLS_DTPREL32	41
+#define	R_68K_TLS_TPREL32	42
 
 typedef unsigned long elf_greg_t;
 
